helix-core/src/main/java/org/apache/helix/monitoring/mbeans/dynamicMBeans/DynamicMBeanProvider.java

```diff
@@ -186,7 +186,7 @@ public AttributeList getAttributes(String[] attributes) {
   }
 
   @Override
-  public MBeanInfo getMBeanInfo() {
+  public synchronized MBeanInfo getMBeanInfo() {
```

Comment:
As we discussed, this sync won't work. Let's just lock the attributes list in the direct access method.

Reply:
How about locking a separate object for both _MBeanInfo and _attributeMap? Once one of them is accessed, we lock both. In this case, we could make sure MBeanInfo and attributes are bound together. 
```
private final Object lockObject = new Object();

synchronized (lockObject) {
}